src/Raven.Server/Documents/Indexes/Static/MapIndex.cs

```diff
@@ -213,6 +214,16 @@ private static MapIndex CreateIndexInstance(IndexDefinition definition, RavenCon
             var staticIndex = IndexCompilationCache.GetIndexInstance(definition, configuration, indexVersion);
 
             var staticMapIndexDefinition = new MapIndexDefinition(definition, staticIndex.Maps.Keys, staticIndex.OutputFields, staticIndex.HasDynamicFields, staticIndex.CollectionsWithCompareExchangeReferences.Count > 0, indexVersion);
+            
+            if (configuration.Indexing.StaticIndexingEngineType is SearchEngineType.Corax)
```

Comment:
We need to have a single place where we'll be able to assert that for all types of indexes. I checked the code and think that `IndexCompilationCache.GenerateIndex()` would be good place to check that.

Reply:
Also please make sure that we have tests which explicitly fail on that. So if we add support for those features then they are supposed to fail.

Reply:
All done. Tests at `FastTests.Corax.StaticIndexes`